### PR TITLE
Implements Forgot password functionality

### PR DESCRIPTION
**Pull request by Hamza-Chaudhary65**
Implements Forgot password functionality

### CODE REVIEW

**Review by ZAINABFATIMA0 — APPROVED** [review]

File: fintrack/users/models.py [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
 from django.db import models
+from django.utils import timezone
 
 
 class User(AbstractUser):
     address = models.CharField(max_length=250)
     email = models.EmailField(unique=True, blank=False)
+
+
+class PasswordReset(models.Model):
```
Comment (sheralim012) [inline fintrack/users/models.py:13]: shouldn't this be connected to user, as user model already has the email field?

File: fintrack/users/models.py [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+    email = models.EmailField()
+    token = models.CharField(max_length=100)
+    expiry_time= models.DateTimeField()
```
Comment (sheralim012) [inline fintrack/users/models.py]:
```suggestion
    expiry_time = models.DateTimeField()
```

File: fintrack/users/models.py [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+
+    def save(self, *args, **kwargs):
+        if not self.id: 
+            self.expiry_time = timezone.now() + timedelta(minutes=10)
```
Comment (sheralim012) [inline fintrack/users/models.py]: you can do this via default value for the field itself

File: fintrack/users/templates/reset_password_email.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<head>
+    <meta charset="UTF-8">
+    <title>Reset Your Password</title>
+</head>
+<body>
+    <div class="container">
+        <h2>Reset Your Password</h2>
+        <p>Hello,</p>
+        <p>We received a request to reset your password. Click the button below to reset it:</p>
+        <p><a href=" {{ reset_link}}" class="button">Reset Password</a></p>
```
Comment (sheralim012) [inline fintrack/users/templates/reset_password_email.html]:
```suggestion
        <p><a href=" {{ reset_link }}" class="button">Reset Password</a></p>
```

File: fintrack/users/v4/views.py [hunk 5]
```diff
@@ -1,10 +1,66 @@
...
+        serializer.is_valid(raise_exception=True)
+
+        user = get_object_or_404(User, email=request.data['email'])
+        
+        token_generator = PasswordResetTokenGenerator()
+        token = token_generator.make_token(user)
+        
+        PasswordReset.objects.create(email=request.data['email'], token=token)
+
+        reset_url = f"{'http://127.0.0.1:8000/api/v4/user/password-reset/'}{token}"
```
Comment (sheralim012) [inline fintrack/users/v4/views.py]: I believe base URL should come from env variable

File: fintrack/users/v4/views.py [hunk 6]
```diff
@@ -1,10 +1,66 @@
...
+
+        send_password_reset_email.delay(self.request.data['email'], reset_url)
+        return Response({"success": "Email sent with forget password link"})
+    
+
+class ResetPasswordConfirm(CreateAPIView):
```
Comment (sheralim012) [inline fintrack/users/v4/views.py]: we aren't creating anything so why use createapiview?

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**